warehouse/api.go

```diff
@@ -874,20 +875,20 @@ func validateObjectStorage(ctx context.Context, request *ObjectStorageValidation
 	return nil
 }
 
-func getFileManagerSettings(provider string, inputConfig map[string]interface{}) *filemanager.SettingsT {
+func getFileManagerSettings(ctx context.Context, provider string, inputConfig map[string]interface{}) *filemanager.SettingsT {
 	settings := &filemanager.SettingsT{
 		Provider: provider,
 		Config:   inputConfig,
 	}
 
-	overrideWithEnv(settings)
+	overrideWithEnv(ctx, settings)
 	return settings
 }
 
 // overrideWithEnv overrides the config keys in the fileManager settings
 // with fallback values pulled from env. Only supported for S3 for now.
-func overrideWithEnv(settings *filemanager.SettingsT) {
-	envConfig := filemanager.GetProviderConfigFromEnv(context.TODO(), settings.Provider)
+func overrideWithEnv(ctx context.Context, settings *filemanager.SettingsT) {
```

Comment:
The context seems to be passed to `backendConfig.WaitForConfig()` only. What if it gets canceled or it times out though?  Do we want `GetProviderConfigFromEnv` to return an error?

Reply:
Do we even need a context propagated to this function?

Reply:
I guess we do, in case of cancelation it would stop waiting for the backend config which is a blocking call, right?

Reply:
Yeah true but that part of code kinda feels weird to me and probably something which no one would ever use.

Because from the UI he won't even be able to create a S3 destination without filling the required fields but we are waiting for config to get the workspaceID for a destination which isn't there. Not sure if I put it out here properly

Reply:
@fracasula I am also aligned with passing the context.